### ISSUE TITLE
Use variables check fails when no variables are defined

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#462]
**Description of the issue**

The rule `Azure.Template.UseVariables` fails when no variables are defined. This also applies to parameters.

**Expected behaviour**

The rule `Azure.Template.UseVariables` should pass when no variables are defined.

For example:

```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "variables": { }
}
```

```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0"
}
```

**Error output**

The following reason is reported:

```text
- The variable '{0}' was not used within the template.
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.15.0-B2008026**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Microsoft Windows 10.0.19041
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
